airbrakes/data_handling/data_processor.py

```diff
@@ -15,9 +15,11 @@ class IMUDataProcessor:
     :param data_points: A sequence of EstimatedDataPacket objects to process.
     """
 
-    __slots__ = ("_avg_accel", "_avg_accel_mag", "_data_points", "_max_altitude")
+    __slots__ = ("_avg_accel", "_avg_accel_mag", "_data_points", "_max_altitude", "upside_down")
+
+    def __init__(self, data_points: Sequence[EstimatedDataPacket], upside_down: bool = False):
+        self.upside_down = upside_down
```

Comment:
So my idea is that anytime a state needs data, it will ask the context and the context will get it from the data processor. So when doing rolling averages or anything else the states need, apply it there